docs/android/features/base-component/customize-views.md

```diff
@@ -0,0 +1,348 @@
+import ai.mealz.sdk.components.baseComponent.myMealButton.success.MyMealButtonSuccessParameters
+
+
+class MyMyMealButtonSuccess: MyMealButtonSuccess {
+
+     @Composable
+    override fun Content(params: MyMealButtonSuccessParameters) {
+        // Your custom design here 
+    }
+}
+```
+with 
+
+```kotlin
+data class MyMealButtonSuccessParameters(
+    val recipeCount: Int, // number or recipe in mealz basket
+    val onClick: () -> Unit // on click action that should redirect to My meal
+)
+```
+
+#### MyMealButton all together
+
+Final step is to set Template in your template configuration: 
+
+```kotlin
+class MealzTemplateManager {
+    init {
+        MiamTheme.Template {
+            defaultViews {
+                myMealButton {
+                    success {
+                        view = MyMyMealButtonSuccess()
+                    }
+                    empty {
+                        view = MyMyMealButtonEmpty()
+                    }
+                }
+            }
+        }
+    }
+}
+```
+
+:::tip 
+Your not force to override all template availble, here you can override only empty state if you wish to.
+:::
+
+## Page States
+
+### Empty
+<BaseComponentView.Empty platform="android"/>
+
+In Order to create your own empty state you have to create class that implement `EmptyPage`
+
+```kotlin
+import ai.mealz.sdk.components.baseComponent.emptyPage.EmptyPage
+import ai.mealz.sdk.components.baseComponent.emptyPage.EmptyPageParameters
+
+
+class MyCustomEmptyView: EmptyPage {
+
+     @Composable
+    override fun Content(params: EmptyPageParameters) {
+        // Your custom design here 
+    }
+}
+```
+
+Then you can use all arguments we give you thinks to `EmptyPageParameters`
+
+```kotlin
+data class EmptyPageParameters(
+    val title: String, // Top sentence
+    val subtitle: String = "", // optional sub sentence
+    val haveAnAction: Boolean = false, // show CTA or not
+    val image: Int = miamEmpty, 
+    val actionText: String = "", // CTA text
+    val action: () -> Unit = {} // onClick on CTA
+)
+```
+
+Final step is to set Template in your template configuration: 
+
+```kotlin
+class MealzTemplateManager {
+    init {
+        MiamTheme.Template {
+            defaultViews {
+                empty {
+                    view = MyCustomEmptyView()
+                }
+             }
+        }
+    }
+}
+```
+
+### Loading
+<BaseComponentView.Loading platform="android"/>
+
+In Order to create your own loading state you have to create class that implement `Loader`
+
+```kotlin
+import ai.mealz.sdk.components.baseComponent.loader.Loader
+import ai.mealz.sdk.components.baseComponent.loader.LoaderParameters
+
+
+class MyCustomLoadingView: Loader {
+
+     @Composable
+    override fun Content(params: LoaderParameters) {
+        // Your custom design here 
+    }
+}
+```
+
+`LoaderParameters` is an empty object for now, but tomorrow we'll be able to add properties to it, in order to avoid breaks in future versions. 
```

Comment:
```suggestion
`LoaderParameters` is an empty object for now, but tomorrow we'll be able to add properties to it, in order to avoid breaking changes in future versions. 
```